src/type/validate.js

```diff
@@ -314,94 +317,110 @@ function validateFields(
   }
 }
 
-function validateObjectInterfaces(
+function validateInterfaces(
   context: SchemaValidationContext,
-  object: GraphQLObjectType,
+  implementing: GraphQLObjectType | GraphQLInterfaceType,
 ): void {
   const implementedTypeNames = Object.create(null);
-  for (const iface of object.getInterfaces()) {
-    if (!isInterfaceType(iface)) {
+  for (const implemented of implementing.getInterfaces()) {
+    if (!isInterfaceType(implemented)) {
       context.reportError(
-        `Type ${inspect(object)} must only implement Interface types, ` +
-          `it cannot implement ${inspect(iface)}.`,
-        getAllImplementsInterfaceNodes(object, iface),
+        `Type ${inspect(implementing)} must only implement Interface types, ` +
+          `it cannot implement ${inspect(implemented)}.`,
+        getAllImplementsInterfaceNodes(implementing, implemented),
       );
       continue;
     }
 
-    if (implementedTypeNames[iface.name]) {
+    if (implementedTypeNames[implemented.name]) {
       context.reportError(
-        `Type ${object.name} can only implement ${iface.name} once.`,
-        getAllImplementsInterfaceNodes(object, iface),
+        `Type ${implementing.name} can only implement ${implemented.name} once.`,
+        getAllImplementsInterfaceNodes(implementing, implemented),
       );
       continue;
     }
-    implementedTypeNames[iface.name] = true;
-    validateObjectImplementsInterface(context, object, iface);
```

Comment:
It's better to use `validateTypeImplementsInterface`.

Reply:
I'm not 100% sure I'm following here (this removes those lines, and `validateTypeImplementsInterface` doesn't exist).

Reply:
I mean if we rename `validateObjectImplementsInterface` to something `validateTypeImplementsInterface` is better follows existing naming schema than `validateImplementsAncestors`

Reply:
Ah, gotcha. As these are separate functions with the same signature, should these both be changed from:

```js
validateImplementsAncestors(context, type, iface);
validateImplementsInterface(context, type, iface);
```

to:
```js
validateTypeImplementsAncestors(context, type, iface);
validateTypeImplementsInterface(context, type, iface);
```

Reply:
(Previously `validateImplementsAncestors ` was only called from inside `validateImplementsInterface`, but this was changed to avoid an extra level of depth in the stack.)

Reply:
I've gone ahead and renamed these functions and variable names; hopefully these look better.